qa/acceptance/spec/shared_examples/cli/logstash-plugin/install.rb

```diff
@@ -0,0 +1,59 @@
+# encoding: utf-8
+require_relative "../../../spec_helper"
+require "logstash/version"
+require "fileutils"
+require "logstash/devutils/rake"
+
+shared_examples "logstash install" do |logstash|
+  before(:each) do
+    logstash.install(LOGSTASH_VERSION)
+  end
+
+  after :each do
+    logstash.uninstall
+  end
+
+  describe "on #{name}" do
+    context "with a direct internet connection" do
+      context "when the plugin exist" do
+        context "from a local `.GEM` file" do
+          let(:gem_name) { "logstash-filter-qatest-0.1.1.gem" }
+          let(:gem_path_on_vagrant) { "/tmp/#{gem_name}" }
+          before(:each) do
+            logstash.download("https://rubygems.org/gems/#{gem_name}", gem_path_on_vagrant)
+          end
+
+          after(:each) { logstash.delete_file(gem_path_on_vagrant) }
+
+          it "successfully install the plugin" do
+            command = logstash.run_command_in_path("bin/logstash-plugin install #{gem_path_on_vagrant}")
```

Comment:
is not that much about hiding, but about creating an abstraction that make
this more DRY. but I agree we can do it later.

- purbon

On Thu, May 19, 2016 at 4:12 PM Pier-Hugues Pellerin <
notifications@github.com> wrote:

> In qa/acceptance/spec/shared_examples/cli/logstash-plugin/install.rb
> <https://github.com/elastic/logstash/pull/5259#discussion_r63885211>:
>
> > +  end
> > +
> > +  describe "on #{name}" do
> > +    context "with a direct internet connection" do
> > +      context "when the plugin exist" do
> > +        context "from a local `.GEM` file" do
> > +          let(:gem_name) { "logstash-filter-qatest-0.1.1.gem" }
> > +          let(:gem_path_on_vagrant) { "/tmp/#{gem_name}" }
> > +          before(:each) do
> > +            logstash.download("https://rubygems.org/gems/#{gem_name}", gem_path_on_vagrant)
> > +          end
> > +
> > +          after(:each) { logstash.delete_file(gem_path_on_vagrant) }
> > +
> > +          it "successfully install the plugin" do
> > +            command = logstash.run_command_in_path("bin/logstash-plugin install #{gem_path_on_vagrant}")
>
> Maybe in the future, but I really like how they match the actual usage of
> the logstash cli tool, not sure if it would be a good thing to hide it
> behind an helper.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/5259/files/48b38104c66c0ab5590df3d39c99451e401478e1#r63885211>
>